mesonbuild/interpreter.py

```diff
@@ -2399,6 +2402,9 @@ def func_declare_dependency(self, node, args, kwargs):
         deps = extract_as_list(kwargs, 'dependencies', unholder=True)
         compile_args = mesonlib.stringlistify(kwargs.get('compile_args', []))
         link_args = mesonlib.stringlistify(kwargs.get('link_args', []))
+        variables = kwargs.get('variables', {})
+        if not isinstance(variables, dict):
+            raise InterpreterException('variables must be a dict.')
```

Comment:
Shouldn't we enforce values are strings here? AFAIK with pkg-config and cmake values cannot be anything else.

Reply:
probably true.